AirLib/include/common/AirSimSettings.hpp

```
@@ -1040,6 +1083,129 @@ struct AirSimSettings {
+        lidar_setting->horizontal_rotation_frequency = settings_json.getInt("RotationsPerSecond", lidar_setting->horizontal_rotation_frequency);
+        lidar_setting->draw_debug_points = settings_json.getBool("DrawDebugPoints", lidar_setting->draw_debug_points);
+
+        lidar_setting->vertical_FOV_upper = settings_json.getFloat("VerticalFOVUpper", lidar_setting->vertical_FOV_upper);
+        lidar_setting->vertical_FOV_lower = settings_json.getFloat("VerticalFOVLower", lidar_setting->vertical_FOV_lower);
+
+        lidar_setting->position = createVectorSetting(settings_json, lidar_setting->position);
+        lidar_setting->rotation = createRotationSetting(settings_json, lidar_setting->rotation);
+    }
+
+    static std::unique_ptr<SensorSetting> createSensorSetting(
+        SensorBase::SensorType sensor_type, const std::string sensor_name,
+        bool enabled)
+    {
+        std::unique_ptr<SensorSetting> sensor_setting;
+
+        switch (sensor_type) {
+        case SensorBase::SensorType::Barometer:
+            sensor_setting = std::unique_ptr<SensorSetting>(new BarometerSetting());
+            break;
+        case SensorBase::SensorType::Imu:
+            sensor_setting = std::unique_ptr<SensorSetting>(new ImuSetting());
+            break;
+        case SensorBase::SensorType::Gps:
+            sensor_setting = std::unique_ptr<SensorSetting>(new GpsSetting());
+            break;
+        case SensorBase::SensorType::Magnetometer:
+            sensor_setting = std::unique_ptr<SensorSetting>(new MagnetometerSetting());
+            break;
+        case SensorBase::SensorType::Distance:
+            sensor_setting = std::unique_ptr<SensorSetting>(new DistanceSetting());
+            break;
+        case SensorBase::SensorType::Lidar:
+            sensor_setting = std::unique_ptr<SensorSetting>(new LidarSetting());
+            break;
+        default:
+            break;
```

I think default here should be to raise exception for unknown sensor_type